frontend/src/components/avatarUpload.js

```diff
@@ -28,10 +28,10 @@ class AvatarUpload extends Component {
       } else {
         this.setState({ status: "failure" });
       }
-      setTimeout(() => {
-        this.setState({ status: "waiting" });
-      }, 2000);
     });
+    setTimeout(() => {
+      this.setState({ status: "waiting" });
+    }, 2000);
   }
```

Comment:
could you revert this change? realizing it'll make legit things break. I'll think abt what else you could do to mitigate...